docs/scarpet/api/Auxiliary.md

```diff
@@ -142,6 +142,33 @@ Available shapes:
      * `height` - height of the cyllinder, defaults to `0`, so flat disk. Can be negative.
      * `level` - level of details, see `'sphere'`.
 
+ * `'poly'`:
+   * Required attributes:
+     * `points` - list of points defining vertices of the polygon
+   * Optional attributes:
+     * `relative` - list of bools. vertices of the polygon that affected by 'follow'. Default means that every point is affacted.
+     * `mode` - how those points are connected. may be "polygon"(default),"strip" or "triangles". "polygon" means that it will be viewed as vertices of a polygon center on the first one. "strip" means that it will be viewed as a triangles strip. "triangles" means that it will be viewed as some triangles that are not related to each other (therefor length of `points` in this mode have to be a multiple of 3).
+     * `inneredges` - if `true` it will make the inner edges be drawn as well. 
+     * `doublesided` - if `true` it will make the shapes visible from the back as well. Default is `true`. 
+     * `needf3b` - if True, can toggle visibility by using F3+B. just like entities' hitbox.
+
+ * `'block_displayer'` \ `'item_displayer'`:
+   * Required attributes:
+     * `pos` - position of the object.
+     * `item` - the object to show. could have NBT. is a item tuple or a string-id (exclusive to `'item_displayer'`)
```

Comment:
option still could be shared between 'block' and 'item'. call it 'id' or 'value', or 'name' or something else. 'id' feels best because is also very short and saves bandwidth, but you may find other arguments for or agains.

Reply:
ah.... i ve already wroten block and item argument..... which support id state and nbt...

Reply:
and another difficulty is block could accept a block value, but item cannot. if they have the same name, then it have to share the same value type.....